test/unittests/test_excel2xml.py

```diff
@@ -0,0 +1,583 @@
+            " fierst sobnode , sekond sobnode , Third Node Ov Testliest",
+            "completely wrong spelling variant of 'first subnode'"
+        ]
+        corrections = {"completely wrong spelling variant of 'first subnode'": "first subnode"}
+        testlist_mapping_returned = excel2xml.create_json_excel_list_mapping(
+            path_to_json="testdata/test-project-systematic.json",
+            list_name="testlist",
+            excel_values=excel_column,
+            sep=",",
+            corrections=corrections
+        )
+        testlist_mapping_expected = {
+            "first noude ov testlist": "first node of testlist",
+            "sekond noude ov testlist": "second node of testlist",
+            "fierst sobnode": "first subnode",
+            "sekond sobnode": "second subnode",
+            "Third Node Ov Testliest": "third node of testlist",
+            "third node ov testliest": "third node of testlist",
+            "completely wrong spelling variant of 'first subnode'": "first subnode"
+        }
+        self.assertDictEqual(testlist_mapping_returned, testlist_mapping_expected)
+
+
+    def test_create_json_list_mapping(self) -> None:
+        testlist_mapping_returned = excel2xml.create_json_list_mapping(
+            path_to_json="testdata/test-project-systematic.json",
+            list_name="testlist",
+            language_label="en"
+        )
+        testlist_mapping_expected = {
+            "First node of the Test-List": "first node of testlist",
+            "first node of the test-list": "first node of testlist",
+            "First Sub-Node": "first subnode",
+            "first sub-node": "first subnode",
+            "Second Sub-Node": "second subnode",
+            "second sub-node": "second subnode",
+            "Second node of the Test-List": "second node of testlist",
+            "second node of the test-list": "second node of testlist",
+            "Third node of the Test-List": "third node of testlist",
+            "third node of the test-list": "third node of testlist"
+        }
+        self.assertDictEqual(testlist_mapping_returned, testlist_mapping_expected)
+
+
+    def test_excel2xml(self) -> None:
+        with open("testdata/excel2xml-expected-output.xml") as f:
+            expected = f.read()
+        for ext in ["xlsx", "xls", "csv"]:
+            excel2xml.excel2xml(f"testdata/excel2xml-testdata.{ext}", "1234", "excel2xml-output")
+            with open("excel2xml-output-data.xml") as f:
+                returned = f.read()
+                self.assertEqual(returned, expected, msg=f"Failed with extension {ext}")
+        if os.path.isfile("excel2xml-output-data.xml"):
+            os.remove("excel2xml-output-data.xml")
+
+    @pytest.mark.filterwarnings("ignore")
+    def test_excel2xml_sample_script(self) -> None:
+        with open("docs/assets/templates/excel2xml_script.py") as f:
+            template_script = f.read()
+            exec(template_script, {})
```

Comment:
I thought it would be good to test the sample script offered for download. But could it be a security risk to execute code from another file?

Reply:
I don't think that's an issue